### ISSUE TITLE
(Linux Ubuntu 14.04) java.awt.IllegalComponentStateException: component must be showing on the screen to determine its location 

### ISSUE THREAD

**Issue by kulik** [LGoodDatePicker/LGoodDatePicker#46]
I have following exception throwed using your library. It's happened while i am trying to pick some date, or time. After applying date or time it is doesn't shown in the field. Same stuff with your demo application. I use Linux Ubuntu 14.04 java 8
➜  Downloads java -version
java version "1.8.0_111"
Java(TM) SE Runtime Environment (build 1.8.0_111-b14)
Java HotSpot(TM) 64-Bit Server VM (build 25.111-b14, mixed mode)

```
Exception in thread "AWT-EventQueue-0" java.awt.IllegalComponentStateException: component must be showing on the screen to determine its location
	at java.awt.Component.getLocationOnScreen_NoTreeLock(Component.java:2064)
	at java.awt.Component.getLocationOnScreen(Component.java:2038)
	at com.github.lgooddatepicker.zinternaltools.TimeMenuPanel.mouseReleasedWhileTimeListIsOpen(TimeMenuPanel.java:90)
	at com.github.lgooddatepicker.zinternaltools.TimeMenuPanel.access$100(TimeMenuPanel.java:20)
	at com.github.lgooddatepicker.zinternaltools.TimeMenuPanel$2.mouseReleased(TimeMenuPanel.java:46)
	at java.awt.AWTEventMulticaster.mouseReleased(AWTEventMulticaster.java:290)
	at java.awt.Component.processMouseEvent(Component.java:6535)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
	at java.awt.Component.processEvent(Component.java:6300)
	at java.awt.Container.processEvent(Container.java:2236)
	at java.awt.Component.dispatchEventImpl(Component.java:4891)
	at java.awt.Container.dispatchEventImpl(Container.java:2294)
	at java.awt.Component.dispatchEvent(Component.java:4713)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4888)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4525)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4466)
	at java.awt.Container.dispatchEventImpl(Container.java:2280)
	at java.awt.Window.dispatchEventImpl(Window.java:2750)
	at java.awt.Component.dispatchEvent(Component.java:4713)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.awt.EventQueue$4.run(EventQueue.java:731)
	at java.awt.EventQueue$4.run(EventQueue.java:729)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)

```

**Comment by BlakeTNC** [comment]
Hello,
Thanks for the report. That's a somewhat strange exception. That function should only be called when the time list is visible, and according to your description it should be visible because you just clicked on it. I don't have any Linux machines for a debugging process, so what I could do at the moment is offer to (slowly) debug the problem myself with your assistance through file transfers, or alternatively rely on you to debug the problem if you have the appropriate programming skills, and submit a pull request with a fix.

I added a piece of code that simply checks to make sure the time list component is visible at the beginning of that function, and exits the function if it is not visible. If you'd like to run the attached demo file, let me know what happens. It could potentially fix the problem, or it could create new problems.

The new code is pasted below:

    private void mouseReleasedWhileTimeListIsOpen() {
        if(!timeList.isVisible()){
            return;
         }

The zipped new demo jar can be downloaded here: 
[LGoodDatePicker-8.2.3-experimental-executable-demo.jar.zip](https://github.com/LGoodDatePicker/LGoodDatePicker/files/640209/LGoodDatePicker-8.2.3-experimental-executable-demo.jar.zip)

Warm regards,
Blake

**Comment by kulik** [comment]
if you have time - it's could be much simplier to chat in skype
kulik_dev

**Comment by BlakeTNC** [comment]
Issue note: The attempted fix that was referenced in the previous message did not solve the problem.

**Comment by BlakeTNC** [comment]
An experimental fix for this issue has been uploaded to the repository. Waiting for involved users and developers to report if the fix is working for them.
-Blake

**Comment by BlakeTNC** [comment]
The involved users and developers have all reported that the fix is working for them. The fix has been released in [LGoodDatePicker 8.3.0](https://github.com/LGoodDatePicker/LGoodDatePicker/releases), and on [Maven Central](http://search.maven.org/#search%7Cga%7C1%7Ca%3A%22LGoodDatePicker%22).
-Blake
